.github/workflows/ci.yml

```diff
@@ -641,14 +642,71 @@ jobs:
       - name: run X509 sample
         run: |
           python3 ${{ env.CI_UTILS_FOLDER }}/run_sample_ci.py --file ${{ env.CI_SAMPLES_CFG_FOLDER }}/ci_run_x509_connect_cfg.json
-      - name: configure AWS credentials (Greengrass)
-        uses: aws-actions/configure-aws-credentials@v2
-        with:
-          role-to-assume: ${{ env.CI_GREENGRASS_ROLE }}
-          aws-region: ${{ env.AWS_DEFAULT_REGION }}
-      - name: run Greengrass Discovery sample
+
+  # Runs the Greengrass samples
+  linux-greengrass-tests:
```

Comment:
Is there any reason why we didnt run it with the linux matrix?

Reply:
This new CI job is based on the `linux-cmoke-tests` job above. By linux matrix you mean specify the image version?

Reply:
My fault, I thought we ran the samples/tests in different images, which it was actually the `linux-smoke-tests`.